src/java/za/co/mip/ablduck/utilities/HTMLGenerator.java

```diff
@@ -0,0 +1,376 @@
+    }
+    
+    private String renderReturns (String returnsType, String returnsDoc) {
+        String returns = "";
+        
+        returns += "<h3 class='pa'>Returns</h3>"
+                   + "<ul>"
+                       + "<li>"
+                           + "<span class='pre'>" + returnsType +"</span>"
+                           + "<div class='sub-desc'>"
+                               + returnsDoc
+                           + "</div>"
+                       + "</li>"
+                    + "</ul>";
+        
+        return returns;
+    }
+    
+    private String renderParams (ParameterObject renderParams) {
+        String param = "";
+        
+        param += "<li>"
+                   + "<span class='pre'>" + renderParams.name + "</span> : "
+                   + renderParams.datatype
+                   + "<div class='sub-desc'>"
+                       + renderParams.comment
+                   + "</div>"
+               + "</li>";
+        
+        return param;
+    }
+    
+    private String renderTags (MetaObject meta) {
+        String tags = "<span class=\"signature\">";
+        
+        Class<?> c = meta.getClass();
+        Field[] fields = c.getDeclaredFields();
+
+        for (Field field : fields) {
+            field.setAccessible(true);
+            Boolean flag = false;
+            String key = null;
+            
+            Annotation[] annotations = field.getDeclaredAnnotations();
+            for(Annotation annotation : annotations){
+                if(annotation instanceof SerializedName){
+                    SerializedName myAnnotation = (SerializedName) annotation;
+                    key = myAnnotation.value();
+                    
+                    try {
+                        if(field.get(meta) instanceof DeprecatedObject || field.get(meta) instanceof String)
+                            flag = true;
+                        else
+                            flag = (Boolean) field.get(meta);
+                    } catch (IllegalAccessException ex) {
+                        throw new BuildException(ex);
+                    }
+                    
+                }
+            }
+            
+            if (flag != null && flag)
+                tags += "<span class='" + key.toLowerCase() + "'>" + key.toUpperCase() + "</span>";
+        }
+        
+        tags += "</span>";
+                
+        return tags;
+    }
+    
+    private String renderDeprecated (String tagname, DeprecatedObject deprecated) {
+        String dep = "<div class='rounded-box deprecated-box deprecated-tag-box'>\n" +
+                         "<p>This " + tagname + " has been <strong>deprected</strong> since " + deprecated.version + "</p>\n" +
+                         deprecated.text + "\n" +
+                     "</div>";
+        return dep;
+    }
+    
+    private String renderInternal (String internal) {
+        String dep = "<div class='rounded-box private-box'>\n" +

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "dep". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1488)